packages/teams-js/rollup.config.mjs

```diff
@@ -6,53 +6,65 @@ import { nodeResolve } from '@rollup/plugin-node-resolve';
 import replace from '@rollup/plugin-replace';
 import terser from '@rollup/plugin-terser';
 import typescript from '@rollup/plugin-typescript';
+import dts from 'rollup-plugin-dts';
 import nodePolyfills from 'rollup-plugin-polyfill-node';
 
 import version from './package.json' assert { type: 'json' };
 
-export default {
-  input: './src/index.ts',
-  output: {
-    dir: 'dist/esm',
-    name: '@microsoft/teams-js',
-    format: 'es',
-    preserveModules: true,
-    entryFileNames: '[name].js',
-    sourcemap: false,
-    plugins: [terser()],
-    globals: {
-      buffer: 'Buffer',
-      tty: 'tty',
-      util: 'util',
-      os: 'os',
+export default [
+  {
+    input: './src/index.ts',
+    output: {
+      dir: 'dist/esm',
+      name: '@microsoft/teams-js',
+      format: 'es',
+      preserveModules: true,
+      entryFileNames: '[name].js',
+      sourcemap: false,
+      plugins: [terser()],
+      globals: {
+        buffer: 'Buffer',
+        tty: 'tty',
+        util: 'util',
+        os: 'os',
+      },
     },
-  },
-  preserveEntrySignatures: 'strict',
-  plugins: [
-    nodeResolve({
-      extensions: ['.js', '.ts', '.d.ts', '.json'],
-    }),
-    replace({
-      preventAssignment: true,
-      'process.env.NODE_ENV': JSON.stringify('production'),
-      PACKAGE_VERSION: JSON.stringify(version.version),
-    }),
-    typescript(),
-    json(),
-    commonjs(),
-    nodePolyfills(),
-  ],
-  treeshake: {
-    moduleSideEffects: [
-      'src/internal/communication.ts',
-      'src/internal/nestedAppAuthUtils.ts',
-      'src/internal/utils.ts',
-      'src/internal/videoEffectsUtils.ts',
-      'src/private/constants.ts',
-      'src/private/interfaces.ts',
-      'src/public/constants.ts',
-      'src/public/handlers.ts',
-      'src/public/interfaces.ts',
+    preserveEntrySignatures: 'strict',
+    plugins: [
+      nodeResolve({
+        extensions: ['.js', '.ts', '.d.ts', '.json'],
+      }),
+      replace({
+        preventAssignment: true,
+        'process.env.NODE_ENV': JSON.stringify('production'),
+        PACKAGE_VERSION: JSON.stringify(version.version),
+      }),
+      typescript(),
+      json(),
+      commonjs(),
+      nodePolyfills(),
     ],
+    treeshake: {
+      moduleSideEffects: [
```

Comment:
Are these the files that can't get tree shaken out?

Reply:
Yes, the bundler should mark those files as containing side-effects, so they should be ignored by the treeshaker. Most of them are references to the `window` object. While allowing them to be treeshaken won't necessarily cause an issue, it is possible that they might. Fortunately, most of these files are frequently used, so there aren't many situations where they would be treeshaken out anyways.

Reply:
Those files are also identified in the package.json, which should instruct the bundler of anyone consuming the library to not treeshake them out either.